pilot/pkg/model/push_context.go

```diff
@@ -820,7 +847,36 @@ func (ps *PushContext) getSidecarScope(proxy *Proxy, workloadLabels labels.Colle
 		}
 	}
 
-	return DefaultSidecarScopeForNamespace(ps, proxy.ConfigNamespace)
+	// We didn't have a Sidecar in the namespace. This means we should use the default - either an implicit
+	// default selecting everything, or pulling from the root namespace.
+	ps.sidecarIndex.defaultSidecarMu.Lock()
+	defer ps.sidecarIndex.defaultSidecarMu.Unlock()
+	if proxy.Type == Router {
+		sc, f := ps.sidecarIndex.gatewayDefaultSidecarsByNamespace[proxy.ConfigNamespace]
+		if f {
+			// We have already computed the scope for this namespace, just fetch it
+			return sc
+		}
+		computed := DefaultSidecarScopeForNamespace(ps, proxy.ConfigNamespace)
+		ps.sidecarIndex.gatewayDefaultSidecarsByNamespace[proxy.ConfigNamespace] = computed
+		return computed
+	}
+	sc, f := ps.sidecarIndex.computedSidecarsByNamespace[proxy.ConfigNamespace]
+	if f {
+		// We have already computed the scope for this namespace, just fetch it
+		return sc
+	}
+	// We need to compute this namespace
+	var computed *SidecarScope
+	if ps.sidecarIndex.rootConfig != nil {
+		computed = ConvertToSidecarScope(ps, ps.sidecarIndex.rootConfig, proxy.ConfigNamespace)
+	} else {
+		computed = DefaultSidecarScopeForNamespace(ps, proxy.ConfigNamespace)
```

Comment:
isn't  this duplicate with L860

Reply:
the code is duplicated but only this or L860 will run, never both

Reply:
Correct, it confuses me with the below line.  We may not have gateways reside in same namespace as normal sidecar proxy